File: LeetCode/src/121.js
// Best Time to Buy and Sell Stock
const maxProfit = function (prices) {
	let buyDay = 0;
	let maxProfit = 0;
	for (let i = 1; i < prices.length; ++i) {
		if (prices[i] > prices[buyDay]) {
			const profit = prices[i] - prices[buyDay];
			maxProfit = Math.max(maxProfit, profit);
		} else {
			buyDay = i;
		}
	}
	return maxProfit;
};

File: LeetCode/src/121.rs
// Best Time to Buy and Sell Stock
use std::cmp;

impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut max_profit = 0;
        let mut buy_day = 0;
        for sell_day in 1..prices.len() {
            if prices[buy_day] < prices[sell_day] {
                let profit = prices[sell_day] - prices[buy_day];
                max_profit = cmp::max(max_profit, profit);
            } else {
                buy_day = sell_day;
            }
        }
        max_profit
    }
}
